pkg/errors/galasaAPIErrorsArray.go

```diff
@@ -24,13 +24,23 @@ func NewGalasaApiErrorsArray(body []byte) (*GalasaAPIErrorsArray, error) {
 
 	errorsGathered := new(GalasaAPIErrorsArray)
 
-	err = json.Unmarshal(body, &jsonArray)
+	//convert payload into string, check the first char to see if it is an array
+	stringBody := string(body)
+	if string(stringBody[0]) == "[" {
```

Comment:
There must be a better way of doing this.
ie: Parsing the json, then asking the json whether it's a string array or an object ?